h/db/query_filters.py

```diff
@@ -0,0 +1,39 @@
+"""A collection of functions to help build queries."""
+
+import operator
+
+import sqlalchemy as sa
+
+
+class BadDateFilter(ValueError):
+    """There is something wrong with the date filter provided."""
+
+
+_DATE_FILTER_OP_MAP = {
+    "gt": operator.gt,
+    "gte": operator.ge,
+    "lt": operator.lt,
+    "lte": operator.le,
+    "eq": operator.eq,
+    "ne": operator.ne,
+}
+
+
+def date_match(column, spec: dict):
+    """
+    Get an SQL comparator for a date column based on dict spec.
+
+    The dict can contain operators as keys and dates as values.
+    :raises BadDateFilter: For unrecognised operators or no spec
+    """
+    if not spec:
+        raise BadDateFilter(f"No spec given to filter '{column}' on")
+
+    clauses = []
+    for op_key, value in spec.items():
+        if op := _DATE_FILTER_OP_MAP.get(op_key):
+            clauses.append(op(column, value))
+        else:
+            raise BadDateFilter(f"Unknown date filter operator: {op_key}")
+
+    return sa.and_(*clauses)
```

Comment:
I think the cost of this probably outweighs the benefit compared to just using sqlalchemy directly in the service. With this I need to understand the date filter map, this function, and the sqlalchemy that it generates (which I can't actually see), so it feels like just layers of obscurity.